### PR TITLE
Add missing resources

### PR DESCRIPTION
**Pull request by fabiomatavelli**
This PR adds the following resources:

- drp_param
- drp_task
- drp_stage
- drp_workflow
- drp_subnet
- drp_reservation
- drp_pool
- drp_profile_param

They were removed on the latest upgrade by RackN.

### CODE REVIEW

File: drpv4/resource_drp_reservation.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+			},
+			"subnet": {
+				Type:        schema.TypeString,
+				Description: "Reservation subnet",
+				Optional:    true,
+			},
+			"token": {
+				Type:        schema.TypeString,
+				Description: "Reservation token",
+				Required:    true,
```
Comment (ihab-soliman-form3) [inline drpv4/resource_drp_reservation.go:88]: Shouldn't this be marked as sensitive?

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,86 @@
...
 
 require (
-	github.com/Bowery/prompt v0.0.0-20190916142128-fa8279994f75 // indirect
-	github.com/VictorLowther/jsonpatch2 v1.0.0
-	github.com/go-test/deep v1.0.3
-	github.com/hashicorp/terraform v0.13.2
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.0.1
-	github.com/pborman/uuid v1.2.0
-	gitlab.com/rackn/provision/v4 v4.8.0-alpha00.0.20210905035228-7afe04a243cf
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.11.1
```
Comment (ihab-soliman-form3) [inline go.mod]: I don't think this is the latest, would it be possible to update?

File: go.mod [hunk 3]
```diff
@@ -1,13 +1,86 @@
 module gitlab.com/rackn/terraform-provider-drpv4
 
-go 1.13
+go 1.18
```
Comment (ihab-soliman-form3) [inline go.mod]: `1.18` is out of support, we should update that

> Each major Go release is supported until there are two newer major releases. For example, Go 1.5 was supported until the Go 1.7 release, and Go 1.6 was supported until the Go 1.8 release. We fix critical problems, including [critical security problems](https://go.dev/security), in supported releases as needed by issuing minor revisions (for example, Go 1.6.1, Go 1.6.2, and so on).

https://go.dev/doc/devel/release

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          go-version-file: go.mod
+      - name: Setup DRP
+        run: |
+          curl -fsSL get.rebar.digital/stable | bash -s -- install --isolated --version=stable
+          ln -s $(pwd)/bin/linux/amd64/dr-provision /usr/local/bin/dr-provision
+          dr-provision --version
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: '0.13.*'
```
Comment (ihab-soliman-form3) [inline .github/workflows/test.yml]: Why use 0.13? this is totally EOL, and we only use 1.3.X now

File: drpv4/resource_drp_reservation.go (hunk 1 above)
Comment (fabiomatavelli-f3) [reply drpv4/resource_drp_reservation.go:88]: It doesn't need to be sensitive as it is not treated as sensitive by DRP.

File: go.mod (hunk 2 above)
Comment (fabiomatavelli-f3) [reply go.mod]: I was able to update to v2.24.1. With v2.25.0 it breaks DRP provision library as it uses some old version of a Hashicorp library that has breaking change.

File: .github/workflows/test.yml (hunk 4 above)
Comment (fabiomatavelli-f3) [reply .github/workflows/test.yml]: I've added tests for Terraform versions from 0.13 until 1.4 now.

File: go.mod (hunk 3 above)
Comment (fabiomatavelli-f3) [reply go.mod]: I've updated to 1.20.

**Review by ihab-soliman-form3 — APPROVED** [review]

**Pull request merged**